src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -723,4 +803,360 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
+				let colIndex = self.getColumnIndex(this._args.column.name);
+				let dataLength: number = self.dataSet.dataRows.getLength();
+
+				// If this is not the "new row" at the very bottom
+				if (activeRow !== dataLength) {
+					currentRow[colIndex] = state;
+					this._textEditor.applyValue(item, state);
+				}
+			}
+
+			isValueChanged(): boolean {
+				return this._textEditor.isValueChanged();
+			}
+
+			validate(): any {
+				let activeRow = self.currentCell.row;
+				let result: any = { valid: true, msg: undefined };
+				let colIndex: number = self.getColumnIndex(this._args.column.name);
+				let newValue: any = this._textEditor.getValue();
+
+				// TODO: It would be nice if we could support the isCellEditValid as a promise
+				if (self.onIsCellEditValid && !self.onIsCellEditValid(activeRow, colIndex, newValue)) {
+					result.valid = false;
+				}
+
+				return result;
+			}
+		}
+
+		return OverridableTextEditor;
+	}
+
+	private getColumnEditor(column: ISlickColumn<any>): any {
+
+		if (column.isEditable === false || typeof column.isEditable === 'undefined') {
+			return undefined;
+		}
+
+		let columnId = column.id;
+		//  let isColumnLoading = this.columnsLoading && this.columnsLoading.indexOf(columnId) !== -1;,
+		// let isColumnLoading = false;
+		let canEditColumn = columnId !== undefined; //&& !isColumnLoading;
+		if (canEditColumn) {
+			return this.getOverridableTextEditorClass();
+		}
+		return undefined;
+	}
+
+	public getColumnIndex(name: string): number {
+		return this._columnNameToIndex[name];
+	}
+
+	private getFormatter = (column: any): any => {
+		return (row, cell, value, columnDef, dataContext) => {
+			let columnId = cell > 0 && this.dataSet.columnDefinitions.length > cell - 1 ? this.dataSet.columnDefinitions[cell - 1].id : undefined;
+			if (columnId) {
+				// let isHighlighted = this.dataSet.highlightedCells && !!this.highlightedCells.find(c => c.row === row && c.column + 1 === cell);
+				// let isColumnLoading = this.columnsLoading && this.columnsLoading.indexOf(columnId) !== -1;
+				// let isShadowed = this.blurredColumns && !!this.blurredColumns.find(c => c === columnId);
+				// let isContext = this.contextColumns && !!this.contextColumns.find(c => c === columnId);
+				let overrideValue = this.overrideCellFn && this.overrideCellFn(row, columnId, value, dataContext);
+				let valueToDisplay = (value + '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
+				let cellClasses = 'grid-cell-value-container';
+				/* tslint:disable:no-null-keyword */
+				let valueMissing = value === undefined || value === null;
+				/* tslint:disable:no-null-keyword */
+				let isOverridden = (overrideValue !== undefined && overrideValue !== null) || (overrideValue !== 'NULL' && row !== this.tables[0].grid.getDataLength() - 1);
+				console.log(isOverridden);
+				if (valueMissing && !isOverridden) {
+					cellClasses += ' missing-value';
+				}
+				// if (isColumnLoading === true && !isOverridden) {
```

Comment:
dead code?

Reply:
Potentially, these were all used in the original version of getFormatter from the slickgrid code but since they weren't used at all it seemed, I kept them just in case if they were needed for any missing functionality, but since they aren't so far, I decided to get rid of them